Update K_J and R_K

realmariano opened the pull request:
To agree to the new SI definitions I suggest to change K_J and R_K to its new standard values. According to the BIPM brochure (https://www.bipm.org/en/publications/mises-en-pratique/), 15 significant digits can be used, this truncation ensures an error of less than 1 part in 10^15 negligible to the majority of possible uses. K_J90 and R_K90 are no longer recommended to be used.

- [ ] Closes # (insert issue number)
- [ ] Executed ``black -t py36 . && isort -rc . && flake8`` with no errors
- [ ] The change is fully covered by automated unit tests
- [ ] Documented in docs/ as appropriate
- [ ] Added an entry to the CHANGES file

hgrecco commented:
Thanks! (Metrologists helping us to keep it correct ;-)

hgrecco commented:
bors r+

hgrecco commented:
Please update the definitions that depend on K_J and R_K when needed removing the `_90`

realmariano commented:
Sorry about that, I shall and send a new pull request.
By the way, thanks for making the project, its great!!!!


Mariano

email: realmariano@gmail.com
skype: real.mariano


On Tue, Jan 7, 2020 at 2:11 AM Hernan Grecco <notifications@github.com>
wrote:

> Please update the definitions that depend on K_J and R_K when needed
> removing the _90
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hgrecco/pint/pull/971?email_source=notifications&email_token=ACXORMG4J47VEVSS3CLBORTQ4QFHXA5CNFSM4KDRNGWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIHW7UQ#issuecomment-571437010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXORMCEPXHUKXHQ3AFCAFDQ4QFHXANCNFSM4KDRNGWA>
> .
>

hgrecco commented:
No problem.

Please also add an entry to CHANGES file indicating the nature of the change and label it as a BREAKING CHANGE (i.e. people using the `_90` definitions will obtain different results or not be able to obtain the units). 

You do not need to make another PR. Just amend the commit in your local computer and force push to your repo. Github will see the change and update this PR

realmariano commented:
Thanks for the help and advice on how to do it in the right way.

I just got into a small hassle: since the new definitions at SI did not
give a different symbol for K_J = 15 digits recommended value number  and
K_J = 2 * e / h  (exact value)  I will keep getting the problem. I will
figure this out and change it as soon as I find how to do it properly.



Mariano

email: realmariano@gmail.com
skype: real.mariano


On Tue, Jan 7, 2020 at 2:27 AM Hernan Grecco <notifications@github.com>
wrote:

> No problem.
>
> Please also add an entry to CHANGES file indicating the nature of the
> change and label it as a BREAKING CHANGE (i.e. people using the _90
> definitions will obtain different results or not be able to obtain the
> units).
>
> You do not need to make another PR. Just amend the commit in your local
> computer and force push to your repo. Github will see the change and update
> this PR
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hgrecco/pint/pull/971?email_source=notifications&email_token=ACXORMH52QSQ3KGJG4XAAJ3Q4QHDXA5CNFSM4KDRNGWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIHXYSQ#issuecomment-571440202>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXORMHHMQUKCWW5HHJBC3DQ4QHDXANCNFSM4KDRNGWA>
> .
>

hgrecco commented:
My feeling is that we want to define it as `K_J = 2 * e / h` and let pint do the math. `e` and `h `are also updated to 2019 in our constants file, and the 15 digits recommended should agree, right?

realmariano commented:
Yes, I agree. Was just looking if there was no issue doing that.


Mariano

email: realmariano@gmail.com
skype: real.mariano


On Tue, Jan 7, 2020 at 2:41 AM Hernan Grecco <notifications@github.com>
wrote:

> My feeling is that we want to define it as K_J = 2 * e / h and let pint
> do the math. e and h are also updated to 2019 in our constants file, and
> the 15 digits recommended should agree, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hgrecco/pint/pull/971?email_source=notifications&email_token=ACXORMCOJKTIAYNLUTH3PGTQ4QIZHA5CNFSM4KDRNGWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIHYT6Y#issuecomment-571443707>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXORMAGORWSGEGEU6N6AX3Q4QIZHANCNFSM4KDRNGWA>
> .
>

realmariano commented:
Dear Hgrecco
As you suggested, I have corrected the changes and placed the advice in the
CHANGES file, also pushed these changes to my forked repo. As I understand
from your previous comment, changes will be caught from my previous PR,
correct? Let me know if you need me to do something else.

Cheers

Mariano

email: realmariano@gmail.com
skype: real.mariano


On Tue, Jan 7, 2020 at 2:43 AM Mariano Real <realmariano@gmail.com> wrote:

> Yes, I agree. Was just looking if there was no issue doing that.
>
>
> Mariano
>
> email: realmariano@gmail.com
> skype: real.mariano
>
>
> On Tue, Jan 7, 2020 at 2:41 AM Hernan Grecco <notifications@github.com>
> wrote:
>
>> My feeling is that we want to define it as K_J = 2 * e / h and let pint
>> do the math. e and h are also updated to 2019 in our constants file, and
>> the 15 digits recommended should agree, right?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/hgrecco/pint/pull/971?email_source=notifications&email_token=ACXORMCOJKTIAYNLUTH3PGTQ4QIZHA5CNFSM4KDRNGWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIHYT6Y#issuecomment-571443707>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACXORMAGORWSGEGEU6N6AX3Q4QIZHANCNFSM4KDRNGWA>
>> .
>>
>

keewis commented:
you pushed to the `master` branch on your own repository. This PR only watches the branch `patch-1`, so you need to update / push that.

hgrecco commented:
Still I am not able to see them. Sorry, please close this PR, apply your changes again to your master branch and make a new PR. I really like the idea of merging this.

realmariano commented:
Sorry, I push it to patch-1 yesterday, seems I did it wrong again. I will
do a new PR.
One thing. Found in a different file with some others definitions _90
dependant. They will show the new values even when they are labeled _90 (eg
conventional power) I will switch then also to the new SI.

On Sun, Jan 12, 2020, 22:09 Hernan Grecco <notifications@github.com> wrote:

> Still I am not able to see them. Sorry, please close this PR, apply your
> changes again to your master branch and make a new PR. I really like the
> idea of merging this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hgrecco/pint/pull/971?email_source=notifications&email_token=ACXORMHAMNZQYLKQB4MCWWDQ5O5K5A5CNFSM4KDRNGWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIXJWLA#issuecomment-573479724>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXORMGXR5BB7V6HGKRSUXDQ5O5K5ANCNFSM4KDRNGWA>
> .
>

keewis commented:
Actually, I think this is not a problem with the PR but with the missing changes: the `K_J90` and `R_K90` have been renamed to `K_J` and `R_K` in `constants_en.txt`, but these are also referenced for units in `default_en.txt`, but I can't find any commits changing `default_en.txt` in any of your branches. A grep for `R_K90` prints:
```
default_en.txt: conventional_watt_90 = K_J90 ** 2 * R_K90 / (K_J ** 2 * R_K) * watt = W_90
default_en.txt: conventional_ampere_90 = K_J90 * R_K90 / (K_J * R_K) * ampere = A_90
default_en.txt: conventional_coulomb_90 = K_J90 * R_K90 / (K_J * R_K) * coulomb = C_90
default_en.txt: conventional_ohm_90 = R_K / R_K90 * ohm = Ω_90 = ohm_90
default_en.txt: conventional_farad_90 = R_K90 / R_K * farad = F_90
default_en.txt: conventional_henry_90 = R_K / R_K90 * henry = H_90
```
and for `K_J90`:
```
default_en.txt: conventional_watt_90 = K_J90 ** 2 * R_K90 / (K_J ** 2 * R_K) * watt = W_90
default_en.txt: conventional_ampere_90 = K_J90 * R_K90 / (K_J * R_K) * ampere = A_90
default_en.txt: conventional_coulomb_90 = K_J90 * R_K90 / (K_J * R_K) * coulomb = C_90
default_en.txt: conventional_volt_90 = K_J90 / K_J * volt = V_90
```
these are the lines that cause the CI failures.

Jellby commented:
I'm not an expert, but my understanding is that K_J and R_K are *exact* constants (since 2019), and they are already defined as such in `constants_en.txt`, although in terms of other constants:

```
josephson_constant = 2 * e / h = K_J
von_klitzing_constant = h / e ** 2 = R_K
```

On the other hand K_J90 and R_K90, while also exact constants, are not defined from other constants, they just have a fixed numerical value and they are used to define "conventional" electrical units, which are slightly different from the SI units, see https://en.wikipedia.org/wiki/Conventional_electrical_unit

Indeed, K_J90 and R_K90 are "deprecated" in the sense that they should not be used in new measurements and reports, but their values do not change, and they should stay for historical or reference purposes.

I think the proposed change is wrong on two accounts:

1. K_J and R_K are not defined from their numerical values, so they should not be in the "DEFINED EXACT CONSTANTS" section.

2. K_J and R_K are not "conventional" constants, they are just the "true" constants, so the name would be wrong.

I believe @realmariano misunderstood the `constants_en.txt` file and the roles of K_J90 and R_K90.

jules-ch commented:
I'm closing this @Jellby pointed that K_J & R_K are already defined with the right constant K_J90 & R_K90 are only there for compatibility.
